test/inductor/test_perf.py

```diff
@@ -343,6 +343,20 @@ def f(a, b, c):
         inp = (T(10, 10), T(10, 10), T(10, 10))
         self.assertExpectedInline(count_numel(f, *inp), """500""")
 
+    def test_cat_pointwise_inps(self):
+        def f(a, b, c):
+            return torch.cat((a + 1, b + 2, c + 3)) + 10
+
+        inp = (T(10, 10), T(10, 10), T(10, 10))
+        self.assertExpectedInline(count_numel(f, *inp), """600""")
+
+    def test_cat_pointwise_inps_non_fused(self):
```

Comment:
I think this kind of test is a little bit worse, since you could imagine in the future we decide to tune the heuristic and we end up fusing it.

I think it would be good to add some more tests for the case where this heuristic *could* trigger but it's suboptimal (i.e. inputs are from a non-fusible op)

Reply:
inputs are from a non-fusible op - you mean where we should be able to memory plan away the cat but we don't ?

Reply:
Actually, I'm thinking about if we have something like `cat([mm(a), mm(b), cos(c)]) + 3`. In this case, I think there would be a total of 8 accesses without this optimization applying, and 6 if it did?

So perhaps it's worth it as long as the downstream operators are fusible?

Reply:
For this case, we would actually apply it. That's the pointwise_uses check.

Reply:
I think you want at least one pointwise input right?